### ISSUE TITLE
Feature request : print only requested columns for CSV/PDF/JSON Export

### ISSUE THREAD

**Issue by gbin2265** [Icinga/icingadb-web#1011]
## Proposal for export's CSV/PDF/JSON

If you now request a CSV export, it provides all the columns that exist.
Wouldn't it be possible to include an additional parameter where you specify which columns you want to see only in the CSV/PDF/EXPORT export?

## Describe the solution you'd like

add parameter like **printcolums**=...

You can then use this parameter as a filter when printing the columns.

### example
**printcolumns**=host.id,host.display_name,host.vars.os,host.address,host.name,host.vars.local_var

### Result
id,display_name,address,name,vars.os,vars.local_var
"1370166c5ea3a961e81ecf3ace578e19dac2184e","host.com","1.1.1.1","obj_host","rhel9","city1"

**Comment by nilmerg** [comment]
Hi,

for PDF this is already possible by using `?columns=` in the hosts and services list. This just needs to be supported for the CSV and JSON export.

**Comment by gbin2265** [comment]
Hi,

Added value was from printcolumns that you can give additional columns like the IDs in the export but not on the screen.

But tell you that the columns option is the fastest/easiest way, I will be very happy already.

**Comment by Nathaniel-Donahue** [comment]
ref/NC/816326

**Comment by nilmerg** [comment]
> This just needs to be supported for the CSV and JSON export.

Actually, I'm not sure what I was talking about here. Exporting e.g. `hosts?columns=…` to CSV/JSON works fine, it just includes not only the declared columns but still all the default columns.

If I export `hosts?columns=hostgroup.name` to JSON, there is a `hostgroup` key with an object `{ "name": … }` in each result.

**Comment by gbin2265** [comment]
If you do an export now, you get all columns in the CSV/JSON.
What I asked was whether it would be possible that, when using the column option, only the specified columns are included in the CSV/JSON export.
Similar to how it already works for the PDF.



what you wrote :
for PDF this is already possible by using ?columns= in the hosts and services list. This just needs to be supported for the CSV and JSON export.

**Comment by nilmerg** [comment]
> only the specified columns are included in the CSV/JSON export. Similar to how it already works for the PDF.

Of course, I just wanted to mention that the basic functionality of this parameter already works.
